src/evaluation/retrieval.py

```diff
@@ -3,27 +3,28 @@
 import torch
 from einops import rearrange
 
-def retrieval_evaluation(model_video, data, multicaption=False):
+def retrieval_evaluation(model_video, data, multicaption=False, device="cuda"):
     if type(data) == dict:
         dataloader = data["val"].dataloader
     else:
         dataloader = data
     model_video.eval()
-    device = "cuda" if torch.cuda.is_available() else "cpu"
+    model_video.to(device, non_blocking=True)
```

Comment:
this has nothing to do with retrieval_evaluation, should be done outside the script